Spelling

jsoref opened the pull request:
Looks like we're going to use this as a tracking PR for a while. I'll split off smaller sets based on recommendations and rebase this as each of merged.

Rounds:
* ~#30326~
* ~#30402~
* ~#30530~
* ~#30531~
* ~#30532~

Follow-ups:
* ~#30677~
* ~#30678~
* ~#30679~
* ~#30680~

Some thoughts:
* This PR is too big (I know this) -- typically large projects ask about splitting in some way -- but I leave the how to the project -- ~it might be~ by top level directory, ~it might be by file type, it might be "comments", then "locals", then "public apis" or something else~
* I keep distinct commits by word family because it makes rebasing / dealing w/ conflicts or dropping things in case they're controversial.
* My general preference is to squash at the very last minute (or let the project do it)
* There are definitely some changes to java final classes -- I expect to be asked to drop these

<!--
Thanks for sending a pull request!  Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
  2. Ensure you have added or run the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][SPARK-XXXX] Your PR title ...'.
  4. Be sure to keep the PR description updated to reflect all changes.
  5. Please write your PR title to summarize what this PR proposes.
  6. If possible, provide a concise example to reproduce the issue for a faster review.
  7. If you want to add a new configuration, please read the guideline first for naming configurations in
     'core/src/main/scala/org/apache/spark/internal/config/ConfigEntry.scala'.
-->

### What changes were proposed in this pull request?
<!--
Please clarify what changes you are proposing. The purpose of this section is to outline the changes and how this PR fixes the issue. 
If possible, please consider writing useful notes for better and faster reviews in your PR. See the examples below.
  1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
  2. If you fix some SQL features, you can provide some references of other DBMSes.
  3. If there is design documentation, please add the link.
  4. If there is a discussion in the mailing list, please add the link.
-->

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).


### Why are the changes needed?
<!--
Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, you can clarify why it is a bug.
-->

Misspelled words make it harder to read / understand content.

### Does this PR introduce _any_ user-facing change?
<!--
Note that it means *any* user-facing change including all aspects such as the documentation fix.
If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
If possible, please also clarify if this is a user-facing change compared to the released Spark versions or within the unreleased branches such as master.
If no, write 'No'.
-->

I believe so. Unfortunately, this PR exceeds GitHub's limits, which means it'll be vaguely hard for everyone. I'll try to leave at least some marks to call some things out.

### How was this patch tested?
<!--
If tests were added, say they were added here. Please make sure to add some test cases that check the changes thoroughly including negative and positive cases if possible.
If it was tested in a way different from regular unit tests, please clarify how you tested step by step, ideally copy and paste-able, so that other reviewers can test and check, and descendants can verify in the future.
If tests were not added, please describe why they were not added and/or why it was difficult to add.
-->

The misspellings have been reported at https://github.com/jsoref/spark/commit/706a726f87a0bbf5e31467fae9015218773db85b#commitcomment-44064356

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/spark/commit/9f88454aa3eb1ff5af0c867fcd619c51d6999188

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

jsoref committed:
spelling: create

Signed-off-by: Josh Soref <jsoref@users.noreply.github.com>

AmplabJenkins commented:
Can one of the admins verify this patch?

HyukjinKwon commented on common/network-common/src/main/java/org/apache/spark/network/util/AbstractFileRegion.java:
```diff
@@ -24,7 +24,7 @@
 
   @Override
   @SuppressWarnings("deprecation")
-  public final long transfered() {
```
This should be kept for compatibility I believe.

jsoref replied:
Yeah (I noticed that as I was reviewing -- https://github.com/apache/spark/pull/30323#discussion_r521050706 -- it takes a long time to write the full review)

AmplabJenkins commented:
Can one of the admins verify this patch?

HyukjinKwon commented on core/src/main/scala/org/apache/spark/util/Utils.scala:
```diff
@@ -1090,20 +1090,20 @@ private[spark] object Utils extends Logging {
     }
     // checks if the hostport contains IPV6 ip and parses the host, port
     if (hostPort != null && hostPort.split(":").length > 2) {
-      val indx: Int = hostPort.lastIndexOf("]:")
```
this is just variable name. I think it's fine to use a short name

jsoref replied:
I imagine that this PR will be split out into at least one more slice (probably after the first split PR is merged).

jsoref commented on docs/sql-ref-syntax-dml-insert-into.md:
```diff
@@ -69,11 +69,11 @@ INSERT INTO students VALUES
++---------+----------------------+----------+
+|     name|               address|student_id|
++---------+----------------------+----------+
+|Amy Smith|123 Park Ave, San Jose|    111111|
```
This is odd. The `INSERT` above doesn't match the output w/o this change.

jsoref commented on docs/sql-ref-syntax-dml-insert-into.md:
```diff
@@ -100,29 +100,29 @@ SELECT * FROM students;
++-------------+--------------------------+---------+
+|         name|                   address|      ssn|
++-------------+--------------------------+---------+
+|Dora Williams|134 Forest Ave, Menlo Park|123456789|
```
This address would actually exist, and the cities are all in the Silicon Valley area, so there's no good reason to misspell this city.

jsoref commented on docs/sql-ref-syntax-qry-select-orderby.md:
````diff
@@ -28,7 +28,7 @@ clause, this clause guarantees a total order in the output.
 
 ```sql
-ORDER BY { expression [ sort_direction | nulls_sort_oder ] [ , ... ] }
+ORDER BY { expression [ sort_direction | nulls_sort_order ] [ , ... ] }
````
This is scary

jsoref commented on examples/src/main/python/sql/arrow.py:
```diff
@@ -285,7 +285,7 @@ def asof_join(l, r):
     print("Running pandas_udf example: Series to Series")
     ser_to_ser_pandas_udf_example(spark)
-    print("Running pandas_udf example: Iterator of Series to Iterator of Seires")
+    print("Running pandas_udf example: Iterator of Series to Iterator of Series")
```
??

jsoref commented on external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala:
```diff
@@ -58,10 +58,10 @@ private[sql] class AvroDeserializer(
 
-  private val dateRebaseFunc = DataSourceUtils.creteDateRebaseFuncInRead(
+  private val dateRebaseFunc = DataSourceUtils.createDateRebaseFuncInRead(
     datetimeRebaseMode, "Avro")
 
-  private val timestampRebaseFunc = DataSourceUtils.creteTimestampRebaseFuncInRead(
+  private val timestampRebaseFunc = DataSourceUtils.createTimestampRebaseFuncInRead(
```
I'm really uncertain about this -- maybe `crete` is a thing??

jsoref commented on launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java:
```diff
@@ -93,7 +93,7 @@
    * when using @link{#start()}.
    */
-  public static final String CHILD_CONNECTION_TIMEOUT = "spark.launcher.childConectionTimeout";
+  public static final String CHILD_CONNECTION_TIMEOUT = "spark.launcher.childConnectionTimeout";
```
This is a public thing...

jsoref commented on licenses-binary/LICENSE-jsp-api.txt:
```diff
@@ -722,7 +722,7 @@ Oracle facilitates your further distribution of this package by adding
 permits you to use that code in combination with other independent
 modules not licensed under the GPLv2.  However, note that this would
-not permit you to commingle code under an incompatible license with
+not permit you to comingle code under an incompatible license with
```
Oops. I try to avoid touching license files -- I'll drop

jsoref commented on mllib/src/main/scala/org/apache/spark/ml/feature/Selector.scala:
```diff
@@ -77,7 +77,7 @@ private[feature] trait SelectorParams extends Params
    */
   @Since("3.1.0")
-  final val fpr = new DoubleParam(this, "fpr", "The higest p-value for features to be kept.",
+  final val fpr = new DoubleParam(this, "fpr", "The highest p-value for features to be kept.",
```
Is this public in some special way?

jsoref commented on project/MimaExcludes.scala:
```diff
@@ -730,7 +730,7 @@ object MimaExcludes {
 
     // [SPARK-22324][SQL][PYTHON] Upgrade Arrow to 0.8.0
-    ProblemFilters.exclude[FinalMethodProblem]("org.apache.spark.network.util.AbstractFileRegion.transfered"),
+    ProblemFilters.exclude[FinalMethodProblem]("org.apache.spark.network.util.AbstractFileRegion.transferred"),
```
?

jsoref commented on project/MimaExcludes.scala:
```diff
@@ -1488,7 +1488,7 @@ object MimaExcludes {
       ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.mllib.rdd.RDDFunctions.treeReduce"),
       ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.mllib.rdd.RDDFunctions.treeAggregate"),
-      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.mllib.tree.configuration.Strategy.defaultStategy"),
+      ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.mllib.tree.configuration.Strategy.defaultStrategy"),
```
?

jsoref commented on python/pyspark/cloudpickle/cloudpickle_fast.py:
```diff
@@ -556,7 +556,7 @@ def dump(self, obj):
         # used `CloudPickler.dispatch` as a class-level attribute storing all
         # reducers implemented by cloudpickle, but the attribute name was not a
-        # great choice given the meaning of `Cloudpickler.dispatch` when
+        # great choice given the meaning of `Cloudpickle.dispatch` when
```
?

jsoref replied:
Hmmm, looks like it should be `CloudPickler.dispatch` and not `Cloudpickler.dispatch`..

jsoref commented on python/pyspark/context.py:
```diff
@@ -258,7 +258,7 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
                 except Exception:
                     warnings.warn(
-                        "Failed to add file [%s] speficied in 'spark.submit.pyFiles' to "
+                        "Failed to add file [%s] specified in 'spark.submit.pyFiles' to "
```
user facing?

jsoref commented on python/pyspark/ml/regression.py:
```diff
@@ -1442,7 +1442,7 @@ def setParams(self, *, featuresCol="features", labelCol="label", predictionCol="
                   maxMemoryInMB=256, cacheNodeIds=False, subsamplingRate=1.0,
                   checkpointInterval=10, lossType="squared", maxIter=20, stepSize=0.1, seed=None,
-                  impuriy="variance", featureSubsetStrategy="all", validationTol=0.01,
+                  impurity="variance", featureSubsetStrategy="all", validationTol=0.01,
```
api?

jsoref commented on python/pyspark/ml/regression.pyi:
```diff
@@ -477,7 +477,7 @@ class GBTRegressor(
         stepSize: float = ...,
         seed: Optional[int] = ...,
-        impuriy: str = ...,
+        impurity: str = ...,
```
api?

jsoref commented on python/pyspark/mllib/clustering.py:
```diff
@@ -843,7 +843,7 @@ def setInitialCenters(self, centers, weights):
     def setRandomCenters(self, dim, weight, seed):
         """
-        Set the initial centres to be random samples from
+        Set the initial centers to be random samples from
```
if this is an en-GB thing, I'll drop (I try not to enforce a particular style; this repo appears very mixed; I think this change was done by an automated pass)

jsoref commented on python/pyspark/mllib/regression.py:
```diff
@@ -739,7 +739,7 @@ def _validate(self, dstream):
         if not self._model:
             raise ValueError(
-                "Model must be intialized using setInitialWeights")
+                "Model must be initialized using setInitialWeights")
```
just a note that this is en-US

jsoref commented on python/pyspark/sql/utils.py:
```diff
@@ -145,10 +145,10 @@ def toJArray(gateway, jtype, arr):
     :param arr: python type list
     """
-    jarr = gateway.new_array(jtype, len(arr))
+    jarray = gateway.new_array(jtype, len(arr))
```
`jarray` is used for the same sort of thing in other files

jsoref commented on resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala:
```diff
@@ -420,7 +420,7 @@ private[spark] object Config extends Logging {
     ConfigBuilder("spark.kubernetes.file.upload.path")
       .doc("Hadoop compatible file system path where files from the local file system " +
-        "will be uploded to in cluster mode.")
+        "will be uploaded to in cluster mode.")
```
user facing

jsoref commented on resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesVolumeUtilsSuite.scala:
```diff
@@ -49,14 +49,14 @@ class KubernetesVolumeUtilsSuite extends SparkFunSuite {
     val sparkConf = new SparkConf(false)
     sparkConf.set("test.persistentVolumeClaim.volumeName.mount.path", "/path")
     sparkConf.set("test.persistentVolumeClaim.volumeName.mount.readOnly", "true")
-    sparkConf.set("test.persistentVolumeClaim.volumeName.options.claimName", "claimeName")
```
??

jsoref commented on resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala:
```diff
@@ -180,7 +180,7 @@ private[spark] class ExecutorPodsAllocator(
     // It's possible that we have outstanding pods that are outdated when dynamic allocation
     // decides to downscale the application. So check if we can release any pending pods early
     // instead of waiting for them to time out. Drop them first from the unacknowledged list,
-    // then from the pending. However, in order to prevent too frequent frunctuation, newly
```
?????

jsoref commented on sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala:
```diff
@@ -566,7 +566,7 @@ object RewriteCorrelatedScalarSubquery extends Rule[LogicalPlan] with AliasHelpe
               case s @ SubqueryAlias(alias, _) =>
                 subqueryRoot = SubqueryAlias(alias, subqueryRoot)
-              case op => sys.error(s"Unexpected operator $op in corelated subquery")
+              case op => sys.error(s"Unexpected operator $op in correlated subquery")
```
user facing

jsoref commented on sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:
```diff
@@ -1116,7 +1116,7 @@ object SQLConf {
   val CODEGEN_FACTORY_MODE = buildConf("spark.sql.codegen.factoryMode")
     .doc("This config determines the fallback behavior of several codegen generators " +
-      "during tests. `FALLBACK` means trying codegen first and then fallbacking to " +
+      "during tests. `FALLBACK` means trying codegen first and then falling back to " +
```
user facing...

jsoref commented on sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala:
```diff
@@ -81,7 +81,7 @@ class UnsupportedOperationsSuite extends SparkFunSuite {
   // Commands
   assertNotSupportedInStreamingPlan(
-    "commmands",
+    "commands",
     DummyCommand(),
     outputMode = Append,
     expectedMsgs = "commands" :: Nil)
```
my change here is probably wrong. would `commands_` be ok?
Often you insert a randomly uppercased letter to indicate you're expecting an error...

jsoref commented on sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala:
```diff
@@ -422,7 +422,7 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(SoundEx(Literal("Moskowitz")), "M232")
     checkEvaluation(SoundEx(Literal("Moskovitz")), "M213")
-    checkEvaluation(SoundEx(Literal("relyheewsgeessg")), "R422")
+    checkEvaluation(SoundEx(Literal("reltheewsgeessg")), "R422")
```
I didn't mean to change this (will drop)

jsoref commented on sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DataTypeParserSuite.scala:
```diff
@@ -124,8 +124,8 @@ class DataTypeParserSuite extends SparkFunSuite {
   unsupported("struct<x int, y string>")
 
   test("Do not print empty parentheses for no params") {
-    assert(intercept("unkwon").getMessage.contains("unkwon is not supported"))
-    assert(intercept("unkwon(1,2,3)").getMessage.contains("unkwon(1,2,3) is not supported"))
```
does this need to be misspelled?

jsoref commented on sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ErrorParserSuite.scala:
```diff
@@ -77,7 +77,7 @@ class ErrorParserSuite extends AnalysisTest {
 
   test("SPARK-21136: misleading error message due to problematic antlr grammar") {
-    intercept("select * from a left joinn b on a.id = b.id", "missing 'JOIN' at 'joinn'")
+    intercept("select * from a left join_ b on a.id = b.id", "missing 'JOIN' at 'join_'")
```
I'm assuming this is ok (it's in the family of the `commmands` thing above)

jsoref commented on sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java:
```diff
@@ -486,7 +486,7 @@ public void readDoubles(
   }
 
-  public void readBinarys(
+  public void readBinaries(
```
this might be a frozen public api...

jsoref commented on sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala:
```diff
@@ -165,7 +165,7 @@ object DataSourceUtils {
   }
 
-  def creteDateRebaseFuncInRead(
+  def createDateRebaseFuncInRead(
```
see earlier about "is `crete` a thing?"

jsoref commented on sql/core/src/test/java/test/org/apache/spark/sql/Java8DatasetAggregatorSuite.java:
```diff
@@ -34,43 +34,43 @@
   public void testTypedAggregationAverage() {
     KeyValueGroupedDataset<String, Tuple2<String, Integer>> grouped = generateGroupedDataset();
-    Dataset<Tuple2<String, Double>> agged = grouped.agg(
+    Dataset<Tuple2<String, Double>> aggregated = grouped.agg(
```
Java doesn't have to be terse...

jsoref commented on sql/core/src/test/resources/sql-tests/inputs/postgreSQL/insert.sql:
```diff
@@ -589,26 +589,26 @@ drop table inserttest;
--- alter table donothingbrtrig_test attach partition donothingbrtrig_test2 for values in (2);
--- insert into donothingbrtrig_test values (1, 'foo'), (2, 'bar');
+-- create table do nothingbrtrig_test (a int, b text) partition by list (a);
+-- create table do nothingbrtrig_test1 (b text, a int);
+-- create table do nothingbrtrig_test2 (c text, b text, a int);
+-- alter table do nothingbrtrig_test2 drop column c;
+-- create or replace function do nothingbrtrig_func() returns trigger as $$begin raise notice 'b: %', new.b; return NULL; end$$ language plpgsql;
+-- create trigger do nothingbrtrig1 before insert on do nothingbrtrig_test1 for each row execute procedure do nothingbrtrig_func();
+-- create trigger do nothingbrtrig2 before insert on do nothingbrtrig_test2 for each row execute procedure do nothingbrtrig_func();
+-- alter table do nothingbrtrig_test attach partition do nothingbrtrig_test1 for values in (1);
+-- alter table do nothingbrtrig_test attach partition do nothingbrtrig_test2 for values in (2);
+-- insert into do nothingbrtrig_test values (1, 'foo'), (2, 'bar');
 -- [SPARK-29386] Copy data between a file and a table
--- copy donothingbrtrig_test from stdout;
+-- copy do nothingbrtrig_test from stdout;
```
oops, didn't mean to do this (will drop)

jsoref commented on sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala:
```diff
@@ -1316,7 +1316,7 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     )
   }
 
-  test("oder by asc by default when not specify ascending and descending") {
```
?

jsoref commented on sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala:
```diff
@@ -783,7 +783,7 @@ class DDLParserSuite extends AnalysisTest with SharedSparkSession {
       "serialization.format" -> "x",
       "line.delim" -> "\n",
-      "colelction.delim" -> "a", // yes, it's a typo from Hive :)
+      "collection.delim" -> "a", // yes, it's a typo from Hive :)
```
oops, will drop

jsoref commented on sql/hive/compatibility/src/test/scala/org/apache/spark/sql/hive/execution/HiveCompatibilitySuite.scala:
```diff
@@ -719,7 +719,7 @@ class HiveCompatibilitySuite extends HiveQueryFileTest with BeforeAndAfter {
     "groupby_multi_insert_common_distinct",
     "groupby_multi_single_reducer2",
     "groupby_multi_single_reducer3",
-    "groupby_mutli_insert_common_distinct",
```
is this a reference to another repo?

jsoref commented on sql/hive/compatibility/src/test/scala/org/apache/spark/sql/hive/execution/HiveCompatibilitySuite.scala:
```diff
@@ -961,8 +961,8 @@ class HiveCompatibilitySuite extends HiveQueryFileTest with BeforeAndAfter {
     "subq2",
     "subquery_exists",
     "subquery_exists_having",
-    "subquery_notexists",
-    "subquery_notexists_having",
```
ibid?

jsoref commented on sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveOptions.scala:
```diff
@@ -108,7 +108,7 @@ object HiveOptions {
     "escapeDelim" -> "escape.delim",
     // The following typo is inherited from Hive...
-    "collectionDelim" -> "colelction.delim",
+    "collectionDelim" -> "collection.delim",
```
will drop

jsoref commented on sql/hive/src/test/resources/ql/src/test/queries/clientpositive/overridden_confs.q:
```diff
@@ -1,4 +1,4 @@
 set hive.exec.post.hooks=org.apache.hadoop.hive.ql.hooks.VerifyOverriddenConfigsHook;
-set hive.config.doesnt.exit=abc;
+set hive.config.does_not.exit=abc;
```
probably need to drop?

jsoref commented:
* I can rebase regularly or infrequently (depends on project preference).
* I can update my changes regularly or only rarely.
-- for large changes like this, the odds of getting conflicts are very high (which is part of why I prefer to work w/ small commits as rebuilding them individually is generally fairly easy)

holdenk commented:
How would you feel about doing this by sub project? We could start in the graph processing where it's less actively developed?

jsoref commented:
Fine by me

jsoref commented:
I presume you mean https://github.com/apache/spark/tree/master/graphx/ ?

srowen reviewed (commented):
Yes I like fixing typos. Breaking this out is OK too. 
We don't want to break any public-facing APIs, not here, and I think you're taking good care not to.
Another concern is simply the code churn interfering with back-ports, cherry-picks, so it's OK to err on the side of not changing things, but, typo fixing is probably worth it.

jsoref commented on R/pkg/tests/fulltests/test_sparkSQL.R:
```diff
@@ -2066,7 +2066,7 @@ test_that("higher order functions", {
     expr("CAST(array(1.0, 2.0, -3.0, -4.0) AS array<double>) xs"),
     expr("CAST(array(0.0, 3.0, 48.0) AS array<double>) ys"),
-    expr("array('FAILED', 'SUCCEDED') as vs"),
+    expr("array('FAILED', 'SUCCEEDED') as vs"),
```
The lack of a second instance of the string. I'm pretty sure it's a misspelling, but usually code changes would be pairwise.

zero323 replied:
The test here is based on equivalence between SQL expressions and the R implementation, so it is data independent.

jsoref commented on core/src/main/resources/org/apache/spark/ui/static/dataTables.rowsGroup.js:
```diff
@@ -166,7 +166,7 @@ RowsGroup.prototype = {
 	},
 	
-	_toogleDirection: function(dir)
+	_toggleDirection: function(dir)
```
Can you list which files I should be excluding?

srowen replied:
Most of the javascript is third-party, except for a few that are probably obviously specific to the Spark UI. That's most of it. There are some copied Hive classes here and there but I don't think you're touching those.

jsoref replied:
I'm going to drop the changes for `core/src/main/resources/org/apache/spark/ui/static/dataTables.rowsGroup.js` in:
3c3a8dc388
0fcc89b7fa

bersprockets commented:
I also noticed ```pushedFilers```, which I would imagine should be ```pushedFilters```:

```bash
bash-3.2$ find . -name "*.scala" | xargs grep 'PushedFilers'
find . -name "*.scala" | xargs grep 'PushedFilers'
./external/avro/src/main/scala/org/apache/spark/sql/v2/avro/AvroScan.scala:    super.getMetaData() ++ Map("PushedFilers" -> seqToString(pushedFilters))
./external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala:           |PushedFilers: \\[IsNotNull\\(value\\), GreaterThan\\(value,2\\)\\]
./sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/csv/CSVScan.scala:    super.getMetaData() ++ Map("PushedFilers" -> seqToString(pushedFilters))
./sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/orc/OrcScan.scala:    super.getMetaData() ++ Map("PushedFilers" -> seqToString(pushedFilters))
./sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScan.scala:    super.getMetaData() ++ Map("PushedFilers" -> seqToString(pushedFilters))
./sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala:            "|PushedFilers: \\[.*\\(id\\), .*\\(value\\), .*\\(id,1\\), .*\\(value,2\\)\\]",
./sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala:            "|PushedFilers: \\[.*\\(id\\), .*\\(value\\), .*\\(id,1\\), .*\\(value,2\\)\\]",
./sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala:            "|PushedFilers: \\[IsNotNull\\(value\\), GreaterThan\\(value,2\\)\\]",
bash-3.2$ 
```

srowen commented:
Hm, could have user-facing implications. I'd leave it (here).

jsoref commented on docs/monitoring.md:
```diff
@@ -421,7 +421,7 @@ to handle the Spark Context setup and tear down.
 In addition to viewing the metrics in the UI, they are also available as JSON.  This gives developers
 an easy way to create new visualizations and monitoring tools for Spark.  The JSON is available for
-both running applications, and in the history server.  The endpoints are mounted at `/api/v1`.  Eg.,
+both running applications, and in the history server.  The endpoints are mounted at `/api/v1`.  For example
```
https://github.com/apache/spark/pull/30323#discussion_r521447840

jsoref commented on sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/OperationState.java:
```diff
@@ -31,7 +31,7 @@
   CLOSED(TOperationState.CLOSED_STATE, true),
   ERROR(TOperationState.ERROR_STATE, true),
-  UNKNOWN(TOperationState.UKNOWN_STATE, false),
+  UNKNOWN(TOperationState.UNKNOWN_STATE, false),
```
Apparently this is a downstream for this: https://github.com/apache/hive/search?q=UKNOWN_STATE, so I'll need to change that first.

I'll drop this shortly. If you have any advice for steps I should take w/ that project, I'm open to advice. Otherwise, I'll do something similar to what I did here (it took a number of days to produce this initial PR).

jsoref replied:
apache/hive#1674 https://github.com/apache/hive/pull/1674/commits/40b81969b47d310363f14fb751e7fb7db7bc10d3

srowen replied:
Yeah I just wouldn't touch the Hive code here for now. Anything non-trivial can be dealt with later.

jsoref replied:
Yeah, in the process of dropping this (currently reviewing the build failures)

jsoref commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertSuite.scala:
```diff
@@ -35,7 +35,7 @@ import org.apache.spark.util.Utils
 case class TestData(key: Int, value: String)
 
-case class ThreeCloumntable(key: Int, value: String, key1: String)
+case class ThreeColumntable(key: Int, value: String, key1: String)
```
Would this be ok?
```suggestion
case class ThreeColumnTable(key: Int, value: String, key1: String)
```

jsoref commented:
So, one thing I noticed, there are a bunch of places where the line length (100 chars) is already violated. My tentative plan is not to fix those (outside of scope). For things where I'm adding a new comment, I'll try to avoid introducing new errors of that category.

jsoref commented on mllib/src/main/scala/org/apache/spark/ml/feature/Binarizer.scala:
```diff
@@ -112,7 +112,7 @@ final class Binarizer @Since("1.4.0") (@Since("1.4.0") override val uid: String)
       }
 
-    val ouputCols = inputColNames.zip(tds).map { case (inputColName, td) =>
+    val mappedOutputCols = inputColNames.zip(tds).map { case (inputColName, td) =>
```
There's an `outputCols` thing a few lines above, the oddly named variable here accidentally didn't collide with it...

jsoref commented:
ok, so, part one is merged. 
`bin`, `repl`, and `streaming` seem to be the next least recently touched...

What's do you recommend doing next?

srowen commented:
Sure. Chunks of 100 files or so, whatever feels logical.

jsoref commented:
Things I know will remain:

* `create` (it spans multiple directories) -- we can address this after the current open PRs resolve (probably using this PR)
* `enabled` (`legacy_setops_precedence_enbled` appears to be a public API -- addressing it would be done as its own distinct PR if at all -- one approach is to add a correct spelling making that the preferred and adding a deprecated annotation to the current spelling -- another approach is to just add a comment acknowledging the API botch)

I think that's everything, but we'll see.

Note to self: I changed one `E.g.` to `For example` (which actually fit w/ a second one that I did the same to earlier), so I'm going to have a merge conflict: resolver=drop.

jsoref commented:
Fwiw, these are the [outstanding changes in this PR](https://github.com/jsoref/spark/compare/spelling-sql-core..jsoref:spelling). I expect this PR to be closed with just one commit `create`. And I'll create a distinct PR for `legacy_setops_precedence_enbled` (`enabled`) and `PushedFilers` (`filters`) which, for the time being, I've added to this series.

jsoref commented:
So, we're left with three commits:
* 0ff10b7ee2a13df02ba2ab5c4c5c86851a531c08 - `create`
* 6cbd74162a77fdc4c04c867952ec42c6e493288b - `enabled`
* d0a58cc5fdb8956f41e2161a5d1f4f6d90c40353 - `filters`

I'm going to move `enabled` and `filters` into distinct PRs because I suspect they're more likely to be API breaks.

This PR can be closed once `create` (the last commit standing here) is merged.

I'm not entirely certain the other two will be merged, and even if they are, they're probably much more complicated than generic `spelling` fixes (as I suspect they'll require shims).

jsoref commented:
I've split out some of the things that I dropped earlier. I'm going to leave the rest alone for now (I've left the ones we might revisit as "unresolved" to make them easier to spot).

jsoref commented:
See comments by @srowen @cloud-fan on 0ff10b7ee2a13df02ba2ab5c4c5c86851a531c08
